src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1058,10 +1046,24 @@ where
                         let mut remap_output = remap_output.activate();
                         let cap = cap_set.delayed(cap_set.first().unwrap());
                         let mut session = remap_output.session(&cap);
-                        session.give(remap_trace_updates);
+                        session.give_vec(&mut batch.updates);
+                    }
+
+                    cap_set.downgrade(batch.upper);
+
+
+                    let mut batch = timestamper.advance().await;
```

Comment:
I still feel like `advance()` is a no-op here because we called `mint()` just before. Could be wrong, though.

Reply:
Yeah it's very likely it's a no-op and in that case the downgrade will be a no-op, but it's better to have it here just in case